ui/src/authentication/services/AuthenticationService.js

```diff
@@ -1,30 +1,77 @@
+import { http } from '@api';
+import { sessionManager } from './SessionManager';
+
 class AuthenticationService {
-  getTokenFromDOM = () => {
-    const element = document.getElementsByName('authorized_token');
-    if (element.length > 0 && element[0].hasAttribute('value')) {
-      return element[0].value;
+  loginWithOauthProvider = (nextUrl, providerUrl) => {
+    localStorage.setItem('ILS_USER_WAS_LOGGEDIN', true);
+    sessionManager.setAnonymous();
+    if (process.env.NODE_ENV === 'production') {
+      window.location =
```

Comment:
Is there a more `react` way to alter `window.location` than directly assign to it?

Reply:
React router but cannot be used outside of component and I think it wouldn't do the window reload part as it is handled internally by the router I think. This is the best way I found and also is the safest in terms of browser compatibility.